src/braket/default_simulator/result_types.py

```diff
@@ -163,6 +168,47 @@ def _(statevector: jaqcd.StateVector):
     return StateVector()
 
 
+class DensityMatrix(ResultType):
+    """
+    Simply returns the given density matrix.
+    """
+
+    def __init__(self, targets: Optional[List[int]] = None):
+        """
+        Args:
+            targets (Optional[List[int]]): The qubit indices on which the reduced density matrix
+                are desired. If no targets are specified, the full density matrix are calculated.
+                Default: `None`
+        """
+        self._targets = targets
+
+    def calculate(self, simulation: DensityMatrixSimulation) -> np.ndarray:
```

Comment:
create a private `singledispatch` function `calculate_density_matrix` with implementations for both `DensityMatrixSimulation` and `StateVectorSimulation`. Then we'll be able to get the density matrix for both types.

Reply:
We don't want to provide density matrix as a result type to state vector simulator right?

Reply:
We can just return | \rho \rangle \langle \rho |, but that can be after launch